driver/web/auth.go

```diff
@@ -15,450 +15,195 @@
-	}
-
-	auth := Auth{services: serviceHandlers, admin: adminHandlers, encAuth: encAuth, bbs: bbsHandlers, tps: tpsHandlers, system: systemHandlers, logger: logger}
-
-	return &auth, nil
-}
-
-// TokenAuthHandlers represents token auth handlers
-type TokenAuthHandlers struct {
-	standard      TokenAuthorization
-	permissions   *PermissionsAuth
-	user          *UserAuth
-	authenticated *AuthenticatedAuth
-}
-
-func (auth *TokenAuthHandlers) start() {
-	auth.standard.start()
-	auth.permissions.start()
-	auth.user.start()
-	auth.authenticated.start()
-}
-
-// newTokenAuthHandlers creates new auth handlers for a
-func newTokenAuthHandlers(auth TokenAuthorization) (*TokenAuthHandlers, error) {
-	permissionsAuth := newPermissionsAuth(auth)
-	userAuth := newUserAuth(auth)
-	authenticatedAuth := newAuthenticatedAuth(*userAuth)
-
-	authWrappers := TokenAuthHandlers{standard: auth, permissions: permissionsAuth, user: userAuth, authenticated: authenticatedAuth}
-	return &authWrappers, nil
-}
-
-// ServicesAuth entity
-type ServicesAuth struct {
-	coreAPIs  *core.APIs
-	tokenAuth *tokenauth.TokenAuth
-	logger    *logs.Logger
-}
+	tpsHandlers := tokenauth.NewHandlers(tpsAuth)
 
-func (auth *ServicesAuth) start() {
-	auth.logger.Info("ServicesAuth -> start")
-}
-
-func (auth *ServicesAuth) check(req *http.Request) (int, *tokenauth.Claims, error) {
-	claims, err := auth.tokenAuth.CheckRequestTokens(req)
+	systemAuth, err := newSystemAuth(serviceRegManager)
 	if err != nil {
-		return http.StatusUnauthorized, nil, errors.WrapErrorAction(typeCheckServicesAuthRequestToken, logutils.TypeToken, nil, err)
+		return nil, errors.WrapErrorAction(logutils.ActionCreate, "system auth", nil, err)
 	}
+	systemHandlers := tokenauth.NewHandlers(systemAuth)
 
-	if claims.Admin {
-		return http.StatusUnauthorized, nil, errors.ErrorData(logutils.StatusInvalid, "admin claim", nil)
-	}
-	if claims.System {
-		return http.StatusUnauthorized, nil, errors.ErrorData(logutils.StatusInvalid, "system claim", nil)
-	}
+	auth := Auth{services: &serviceHandlers, admin: &adminHandlers, enc: &encHandlers, bbs: &bbsHandlers, tps: &tpsHandlers, system: &systemHandlers}
 
-	err = auth.tokenAuth.AuthorizeRequestScope(claims, req)
-	if err != nil {
-		return http.StatusForbidden, nil, errors.WrapErrorAction(typeCheckScope, logutils.TypeRequest, nil, err)
-	}
-
-	return http.StatusOK, claims, nil
+	return &auth, nil
 }
 
-func (auth *ServicesAuth) getTokenAuth() *tokenauth.TokenAuth {
-	return auth.tokenAuth
-}
+// ServicesAuth
 
-func newServicesAuth(coreAPIs *core.APIs, serviceRegManager *authservice.ServiceRegManager, serviceID string, logger *logs.Logger) (*ServicesAuth, error) {
+func newServicesAuth(serviceRegManager *authservice.ServiceRegManager, serviceID string) (*tokenauth.StandardHandler, error) {
 	servicesScopeAuth := authorization.NewCasbinScopeAuthorization("driver/web/authorization_services_policy.csv", serviceID)
```

Comment:
It looks like we are passing the same file as both a scope and permission policy. This issue was not introduced here but we should fix it because it may result in unexpected behavior.